src/components/TreeWalker/README.md

```diff
@@ -0,0 +1,55 @@
+# TreeWalker
+
+This is a set of utilities that together provide functionality to traverse up and down the OpenSRP jurisdictions tree.
+
+## withTreeWalker
+
+`withTreeWalker` is a higher order component that takes a regular component as an input, and adds tree walking super powers to it. :)
+
+```tsx
+interface SomeProps extends WithWalkerProps {
+  smile: string;
+}
+
+const SomeComponent = withTreeWalker<SomeProps>(() => <div>I Love Oov</div>);
+
+// then use it like this:
+<SomeComponent jurisdictionId={someJurisdiction.id} limits={limitTree} smile=":-)" />;
+```
+
+`withTreeWalker` will pass all the original props to the wrapped component (that would be `SomeComponent` in the example above). But, crucially, it adds other props which are the result of jurisdiction tree-walking:
+
+- currentNode: the current jurisdiction
+- currentChildren: the current jurisdiction's children
+- hierarchy: the full path to the current jurisdiction, ordered from the root parent, all the way to the current jurisdiction
+- loadChildren: a callback that when called loads the children of the desired jurisdiction. Usually it will be called with an input being one of the elements in the `currentChildren` array, and would result in tree traversal to view that element's children.
+
+These 4 props keep on changing as you go up and down the jurisdiction tree to allow the wrapped component to react to the changes. Basically, `withTreeWalker` provides any old component the ability to traverse and react to OpenSRP jurisdictions tree traversal in a generic way.
+
+## Helpers
+
+### getAncestors
+
+This function is able to recursively call the OpenSRP location API to get all the ancestors of a given jurisdictions.
+
+```ts
+getAncestors(someJurisdiction);
+// returns something like [immediateParent, grandParent, rootParent]
```

Comment:
I think this is ready for shipping.

I however seem to think the correct order of the the returned ancestors is `[immediateParent, grandParent, rootParent].reverse()`, is this correct?

Reply:
@p-netm you're right.  I changed it.  Please re-review